torch/quantization/fx/_equalize.py

```diff
@@ -229,3 +225,41 @@ def calculate_equalization_scale(input_obs: _InputEqualizationObserver,
     equalization_scale = torch.sqrt((max_weights - min_weights) / (max_inputs - min_inputs))
 
     return equalization_scale
+
+
+class EqualizationQConfig(QConfig):
+    """
+    Describes how to quantize a layer or a part of the network specifically for
+    equalization by providing settings (observer classes) for inputs, outputs,
+    and weights.
+
+    Note that EqualizationQConfig needs to contain observer **classes** (like
+    MinMaxObserver) or a callable that returns instances on invocation, not the
+    concrete observer instances themselves.
+    Quantization function will instantiate observers multiple times for each of
+    the layers.
+
+    Observer classes have usually reasonable default arguments, but they can be
+    overwritten with `with_args` method (that behaves like functools.partial):
+
+    my_qconfig = EqualizationQConfig(input_activation=_InputEqualizationObserver.with_args(dtype=torch.qint8),
+                                    output_activation=MinMaxObserver.with_args(dtype=torch.qint8),
+                                    weight=_WeightEqualizationObserver.with_args(dtype=torch.qint8))
+    """
+    def __new__(cls, input_activation=torch.nn.Identity, output_activation=torch.nn.Identity, weight=torch.nn.Identity):
+        # catch common mistakes
+        if isinstance(input_activation, nn.Module) or isinstance(output_activation, nn.Module) or isinstance(weight, nn.Module):
+            raise ValueError("EqualizationQConfig received observer instance, please pass observer class instead. " +
+                             "Use MyObserver.with_args(x=1) to override arguments to constructor if needed")
+        self = super(EqualizationQConfig, cls).__new__(cls, output_activation, weight)
+        self.input_activation = input_activation
```

Comment:
Should we also assert on the type of input_activation and weight? For equalization we should only support _InputEqualizationObserver and similar for weight

Reply:
I'm not sure how to check the type of the input_activation/weights as they are `functool.partial`s. I could do `isinstance(input_activation(), _InputEqualizationObserver)`, but that would create an `input_activation` instance.